### PR TITLE
grafana-mimir: 2.0.0 -> 2.1.0

### PR DESCRIPTION
**Pull request by Mic92**
###### Description of changes

+ add a test

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@ofborg test grafana-mimir

File: nixos/modules/services/monitoring/mimir.nix [hunk 1]
```diff
@@ -56,6 +56,7 @@ in {
         ProtectSystem = "full";
         DevicePolicy = "closed";
         NoNewPrivileges = true;
+        WorkingDirectory = "/var/lib/mimir";
```
Comment (happysalada) [inline nixos/modules/services/monitoring/mimir.nix:59]: Out of curiosity why did you need this ?

File: pkgs/servers/monitoring/mimir/default.nix [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
-    sha256 = "T1lljhC/TS3eoIc9AFo9Oy9/COM/XnfcwtkU618YCdM=";
+    sha256 = "sha256-n7Vzp/GQIC+Mryu9SycMZ3ScPo5O+5tA4TdigpKzmLU=";
   };
 
   vendorSha256 = null;
 
-  subPackages = [ "cmd/mimir" ];
+  subPackages = [
+    "cmd/mimir"
+    "cmd/mimirtool"
```
Comment (happysalada) [inline pkgs/servers/monitoring/mimir/default.nix:17]: This is a great addition that I missed!

Comment (happysalada) [reply pkgs/servers/monitoring/mimir/default.nix:17]: Should we add environment.systemPackages in the module to make sure the user has access to the mimirtool ?

**Comment by happysalada** [comment]
great minds think alike https://github.com/NixOS/nixpkgs/pull/175110#pullrequestreview-988548520  ! :-P

I'm going to close my PR in favor of yours, I only have minor comments, I don't think we will be long merging this.

File: nixos/modules/services/monitoring/mimir.nix (hunk 1 above)
Comment (Mic92) [reply nixos/modules/services/monitoring/mimir.nix:59]: From the commit message:

> Mimir writes files relative to its working directory.
> With this option less paths have to be configured.

File: pkgs/servers/monitoring/mimir/default.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/servers/monitoring/mimir/default.nix:17]: See https://github.com/NixOS/nixpkgs/pull/175189/commits/373739ea4133e31a4f7963b826fef9d935ec83de

**Comment by happysalada** [comment]
Thank you!
